python/tool/sphinx/rules.bzl

```diff
@@ -0,0 +1,84 @@
+#
+# Copyright (C) 2022 Vaticle
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+def _sphinx_docs_impl(ctx):
+    package = ctx.actions.declare_directory("package")
+
+    ctx.actions.run_shell(
+        inputs = ctx.files.target,
+        outputs = [package],
+        command = 'PACKAGE=$(find . -name "*.tar.gz") && tar -xf ${PACKAGE} && mv */%s %s'
```

Comment:
you can make an empty directory and untar into it directly with `tar -xf <archive> -C <output directory>`, which I think will be cleaner. 

You can then also handle tar.gz that contain a top-level directory inside (`package_subdir`), or ones that don't and expand their files directly. This means your rule can become more generic and handle both cases, and make `package_subdir` optional.